driver/convert.c

```diff
@@ -4255,10 +4257,80 @@ static SQLRETURN struct_to_iso8601_timestamp(esodbc_stmt_st *stmt,
 	return SQL_SUCCESS;
 }
 
+/* apply corrections depending on the (column) size and decimal digits
+ * values given at binding time: nullify or trim the resulted string:
+ * https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/column-size
+ * */
+static SQLRETURN size_decdigits_for_iso8601(esodbc_rec_st *irec,
+	SQLULEN *_colsize, SQLSMALLINT *_decdigits)
+{
+	SQLULEN colsize;
+	SQLSMALLINT decdigits;
+	esodbc_stmt_st *stmt = HDRH(irec->desc)->stmt;
+
+	colsize = get_param_size(irec);
+	DBGH(stmt, "requested column size: %llu.", colsize);
+
+	decdigits = get_param_decdigits(irec);
+	DBGH(stmt, "requested decimal digits: %llu.", decdigits);
+	if (ESODBC_MAX_SEC_PRECISION < decdigits) {
+		WARNH(stmt, "requested decimal digits adjusted from %hd to %d (max).",
+			decdigits, ESODBC_MAX_SEC_PRECISION);
+		decdigits = ESODBC_MAX_SEC_PRECISION;
+	}
+
+	switch (irec->es_type->data_type) {
+		case SQL_TYPE_TIME:
+			if (colsize) {
+				if (colsize < TIME_TEMPLATE_LEN(0) ||
+					colsize == TIME_TEMPLATE_LEN(1) - 1 /* `:ss.`*/) {
+					ERRH(stmt, "invalid column size value: %llu; allowed: "
+						"8 or greater than 9");
```

Comment:
`colsize` is not being passed to populate the `%llu`.

Reply:
Thanks, fixed.